sdk/python/src/sangria_sdk/adapters/fastapi.py

```
@@ -89,3 +99,138 @@ async def wrapper(*args: Any, **kwargs: Any) -> Any:
+                        "error": verify_result.message or "Payment verification failed",
+                        "error_reason": verify_result.reason,
+                    },
+                )
+
+            settle_fn, get_result = merchant_client.create_settle_fn(max_price)
+            kwargs["settle"] = settle_fn
+
+            result = await func(*args, **kwargs)
+
+            if not isinstance(result, Settled):
+                raise TypeError("Sangria: handler must return settle(amount, body)")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)